### PR TITLE
Add exhaustive From<T> impls for Value

### PR DESCRIPTION
**Pull request by SergioBenitez**
As discussed on IRC.

### CODE REVIEW

File: src/value.rs [hunk 1]
```diff
@@ -219,47 +220,62 @@ impl<I> ops::IndexMut<I> for Value where I: Index {
...
+            fn from(val: $T) -> Value {
+                Value::$variant(val $($extra)*)
+            }
+        }
+    )
+}
+
+impl_into_value!(String: String);
+impl_into_value!(Integer: i64);
+impl_into_value!(Integer: isize, as i64);
```
Comment (alexcrichton) [inline src/value.rs]: Could this use `.into()` instead of `as $t` perhaps? That way we can ensure the standard library guarantees lossless conversions.

**Comment by alexcrichton** [comment]
Thanks!

File: src/value.rs (hunk 1 above)
Comment (SergioBenitez) [inline src/value.rs]: That works for everything _but_ `isize`. Should we remove the `isize` conversion?

Comment (alexcrichton) [inline src/value.rs]: Yeah seems ok to me!

Comment (SergioBenitez) [inline src/value.rs]: Great! Took care of it.

**Pull request merged**